### ISSUE TITLE
cdtime.r2c problem

### ISSUE THREAD

**Issue by durack1** [UV-CDAT/cdms#2]
The logic of the `cdtime.r2c` does not appear obvious to me, and [the docs aren't helping much](http://uvcdat.llnl.gov/documentation/cdms/cdms_3.html)

```
import cdtime
cdtime.r2c(int(24178),'months since 0')
Traceback (most recent call last):
  File "<ipython-input-111-0a5bda97b1eb>", line 1, in <module>
    cdt.r2c(int(24178),'months since 0')
TypeError: an integer is required
```
Isn't `24178` and integer?
@doutriaux1 @dnadeau4 any tips here would be appreciated

**Comment by durack1** [comment]
@dnadeau4 @doutriaux1 there is an incredible amount of functionality in `cdtime`, however the logic is not exposed well, the documentation mostly doesn't exist and the errors (as above) are less than helpful - so I would advocate that this is resolved by removing duplication and exposing the logic through the docs

**Comment by doutriaux1** [comment]
@durack1 the required `int` is an int representing a calendar

**Comment by doutriaux1** [comment]
e.g `cdtime.NoLeapCalendar` or similar

**Comment by doutriaux1** [comment]
@dnadeau4 yes i will keep it in its own repo

**Comment by jypeter** [comment]
I completely agree with @durack1 (and I have probably said so many times before) that cdtime is a big asset of UV-CDAT and provides calendar related tools that you can't find in _datetime_ and _numpy datetime64_.

It's a bit off topic here, but I don't have time to investigate more and maybe somebody can answer and start a proper issue if need be: can cdtime functions/etc... work on arrays? eg convert/work on lots of time steps in one call. I kinda remember that I had to use explicit (and inefficient) loops for that. This came up again recently when I was trying to help a colleague working with satellite data. I failed at that and I will have to send my question to the mailing list (when I have time)

**Comment by doutriaux1** [comment]
@jypeter it can't work on arrays but it is a great suggestion, please open a separate issue.

On a side note `axes` in cdms2 have convenience functions:

```python

t = data.getTime()
t.toRelativeTime("days since 1900") # converts units on the whole axes
# it is different from
r = t.asRelativeTime("days since 1900") # which simply returns a list of the relative times
```

**Comment by durack1** [comment]
@doutriaux1 rather than us rant in this `CDMS` repo, want to create a `CDTIME` repo that we can start collating these issue under?

**Comment by doutriaux1** [comment]
https://github.com/UV-CDAT/cdtime

**Comment by durack1** [comment]
@dnadeau4 @doutriaux1 it might be worthwhile that we migrate some of the open `UV-CDAT` issues across to the `CDMS` and `CDTIME` repos.. I know I have a bunch of open issues that are `cdms` focused, and this one in particular should be in `cdtime`.. It does seem that it's relatively straight forward too: https://github.com/holman/ama/issues/413

**Comment by durack1** [comment]
@dnadeau4 we can redirect this issue into https://github.com/UV-CDAT/cdtime - it'd be good to migrate this issue

**Comment by durack1** [comment]
@doutriaux1 @dnadeau4 there does seem to be a **working** version of issue migration at the moment:
```
From: "Antonio Diaz (GitHub Staff)"
Date: Thursday, August 25, 2016 at 5:05 PM
To: Paul Durack
Subject: Re: issue migration across repos

Hey Paul,

Thanks for reaching out! GitHub doesn't currently have a way to move or duplicate issues from one repository to another through the website.

As an alternative, you might consider using a new API endpoint we're working on and which was designed with migrations in mind:

It should allow you to preserve the dates of issues and comments that you migrate.
It doesn't trigger notifications for each issue and comment that's created.
It isn't affected by the content-creation abuse rate limits, which would normally cause trouble if you attempt to create lots of content in a short period of time:
```
[https://developer.github.com/v3/#abuse-rate-limits](https://developer.github.com/v3/#abuse-rate-limits)
```
Note that this API doesn't currently allow you create issues as authored by other users, since that would be a security concern. It's something we're considering adding in the future by allowing users to opt-in to such imports, but I can't say when that will happen. For now, you could add that information in a short header/footer in the content you create (e.g., at the bottom of each issue and comment).

We wrote up a walkthrough of using this API in this gist:
```
[https://gist.github.com/jonmagic/5282384165e0f86ef105](https://gist.github.com/jonmagic/5282384165e0f86ef105)
```
Could you give that a try with a few issues and a test project to see if it fits your needs?

Cheers,
Antonio
GitHub Support

Hi folks..

I just read the thread at 
```
[https://github.com/holman/ama/issues/413](https://github.com/holman/ama/issues/413) 
```
(Transferring issues to another repository in an organization)..

Is there a ‘github endorsed’ way to do this yet?

Cheers,

P
```
